lib/public/ContentManager.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Nextcloud - ContextChat
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Anupam Kumar <kyteinsky@gmail.com>
+ * @copyright Anupam Kumar 2024
+ */
+
+namespace OCA\ContextChat;
+
+use OCA\ContextChat\BackgroundJobs\InitialContentImportJob;
+use OCA\ContextChat\BackgroundJobs\SubmitContentJob;
+use OCA\ContextChat\Db\QueueContentItem;
+use OCA\ContextChat\Db\QueueContentItemMapper;
+use OCA\ContextChat\Service\LangRopeService;
+use OCA\ContextChat\Service\ProviderConfigService;
+use OCP\BackgroundJob\IJobList;
+use OCP\Server;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
+use Psr\Log\LoggerInterface;
+
+class ContentManager {
+	public function __construct(
+		private IJobList $jobList,
+		private ProviderConfigService $configService,
+		private LangRopeService $service,
+		private QueueContentItemMapper $mapper,
+		private LoggerInterface $logger,
+	) {
+	}
+
+	/**
+	 * @param string $providerClass
```

Comment:
```suggestion
	 * @param class-string<IContentProvider> $providerClass
```

This makes psalm / your IDE a bit smarter :)

Reply:
nice!